pagopa/io-app
# feat: [IOCOM-329] FIMS cookie

⚠️ This PR depends on #4700 ⚠️

## Short description
This PR sets the FIMS cookie following the [DR](https://pagopa.atlassian.net/wiki/spaces/FIM/pages/474447886/DR+Federated+Identity+Management+System#Gestione-del-Federation-Cookie) specifications

## List of changes proposed in this pull request
- added selector to retrieve the FIMS token from the user's session
- changed cookie name to `X-IO-FIMS-Token`
- cookie value set to the to FIMS token
- removed redefinition of the Cookie type
- `react-native-cookie/cookies` library updated to latest version
- cookie's attributes dependent on the environment configuration in order to test with `io-dev-api-server`

## How to test
Using the `io-dev-api-server`, change the `withCTA` configuration to generate a message with FIMS CTA and enable the [FIMS dev feature flag](https://github.com/pagopa/io-dev-api-server/blob/d5b746e445d545bb150660b96d6bcfa9132072f0/src/payloads/backend.ts#L45) in `src/payloads/backend.ts`. 
Select such message and tap FIMS CTA at the bottom. A webview should open with a section listing the FIMS cookie's value.
